include/libtorrent/aux_/allocating_handler.hpp

```diff
@@ -61,6 +61,12 @@ namespace libtorrent { namespace aux
 		handler_storage(handler_storage const&);
 	};
 
+	struct error_handler_interface
+	{
+		virtual void on_exception(std::exception&) = 0;
```

Comment:
it should probably be a const reference, just like the catch statements probably should be const reference as well